src/components/Dashboard/Dashboard.tsx

```diff
@@ -1,18 +1,22 @@
-import { ChangeEvent, FormEvent, useState } from 'react';
+import { FormEvent, useState } from 'react';
 import classnames from 'classnames';
 import Image from 'next/image';
 import { useRouter } from 'next/router';
 
-import { Row } from '../Row/Row';
 import { trpc } from '../../utils/trpc';
-import addActivity from './img/plus.png';
-import { UserBox } from './UserBox';
 import { Button } from '../Button/Button';
+import { Row } from '../Row/Row';
+import { TextField } from '../TextField/TextField';
+import { UserBox } from './UserBox';
+import addActivity from './img/plus.png';
 import style from './Dashboard.module.scss';
 
 export const Dashboard = () => {
   const [formVisible, setFormVisible] = useState(false);
   const [newActivity, setNewActivity] = useState('');
+  const [newAmount, setNewAmount] = useState(0);
+  const [newUnit, setNewUnit] = useState('');
```

Comment:
Could we go with the implicit approach? This is similar to the register form, where we use the form ref to access the values.

You could then use form.resetValues instead of setting everything to empty string

Reply:
fixed but only temporarely due to faulty type of ref. Need to be fixed properly.